lib/class-wp-rest-post-search-handler.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Search_Handler class
+ *
+ * @package gutenberg
+ * @since 3.3.0
+ */
+
+/**
+ * Core class representing a search handler for posts in the REST API.
+ *
+ * @since 3.3.0
+ */
+class WP_REST_Post_Search_Handler extends WP_REST_Object_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.3.0
+	 */
+	public function __construct() {
+		$this->type     = 'post';
+		$this->subtypes = array_values( get_post_types( array(
+			'public'       => true,
+			'show_in_rest' => true,
+		), 'names' ) );
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since 3.3.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Object_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Object_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+
+		// Get the post types to search for the current request.
+		$post_types = $request[ WP_REST_Search_Controller::PROP_SUBTYPE ];
+		if ( in_array( WP_REST_Search_Controller::TYPE_ANY, $post_types, true ) ) {
+			$post_types = $this->subtypes;
+		}
+
+		// Get the public post statuses as only those should be searched.
+		$post_statuses = array_values( get_post_stati( array(
+			'public'   => true,
+			'internal' => false,
+		), 'names' ) );
+
+		$query_args = array(
+			'post_type'           => $post_types,
+			'post_status'         => $post_statuses,
+			'paged'               => (int) $request['page'],
+			'posts_per_page'      => (int) $request['per_page'],
+			'orderby'             => 'ID',
+			'order'               => 'DESC',
+			'ignore_sticky_posts' => true,
+			'fields'              => 'ids',
+		);
+
+		// If a search term is given, add it and order by relevance.
```

Comment:
Is there a reason we default to `ID` and then switch to relevance?

Reply:
When `search` is not given, it is impossible to sort by relevance, so we need to have some default.

Reply:
> When `search` is not given, it is impossible to sort by relevance, so we need to have some default.

Makes sense. It'd be great to have this clarified in the comment.